hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordScanner.java

```diff
@@ -281,7 +283,7 @@ private void scanInternalV1(Option<KeySpec> keySpecOpt) {
         final String instantTime = logBlock.getLogBlockHeader().get(INSTANT_TIME);
         totalLogBlocks.incrementAndGet();
         if (logBlock.isDataOrDeleteBlock()) {
-          if (this.tableVersion.lesserThan(HoodieTableVersion.EIGHT)) {
+          if (this.tableVersion.lesserThan(HoodieTableVersion.EIGHT) && !allowInflightInstants) {
```

Comment:
can we ensure we have a tbl v6 test for the following:
MOR w/ log files from pending ingestion commit. when we do snapshot read, the log file from pending ingestion commit should be skipped by the log record reader

Reply:
Addressed